extension/intents/nicknames/routineExecutor.js

```diff
@@ -0,0 +1,48 @@
+/* globals log */
+
+import * as intentRunner from "../../background/intentRunner.js";
+
+export class RoutineExecutor {
+  constructor(routineName, subcommands, index = 0) {
+    this.routineName = routineName;
+    this.subcommands = subcommands;
+    this.index = index;
+  }
+
+  async runSubcommand(subcommand) {
+    let hadError = false;
+    subcommand.onError = () => {
+      hadError = true;
+    };
+
+    subcommand.parentRoutine = {
+      name: this.routineName,
+      nextIndex: this.index + 1,
+      stop: false,
+    };
+    log.info(
+      "  Running subintent",
+      subcommand,
+      subcommand.name,
+      subcommand.slots
+    );
+
+    await intentRunner.runIntent(subcommand);
+    if (hadError) {
+      log.info("  Last intent failed, stopping");
+      return true;
```

Comment:
And then this should probably show the error message and the routine name (i.e., `routine {name} encountered error {error} while running {subcommand.utterance}`